tasks/in-house-ctf/README.md

```diff
@@ -0,0 +1,64 @@
+# In House CTF Tasks
+This directory includes Capture The Flag challenges migrated from Google DeepMind's [Dangerous Capabilities Evaluations](https://github.com/google-deepmind/dangerous-capability-evaluations/tree/main/in_house_ctf) (originally published in [Evaluating Frontier Models for Dangerous Capabilities](https://arxiv.org/pdf/2403.13793)). Here we include 9 of the 13 In-House CTF challenges (excluding 4 challenges that have faulty Docker images from the original repository). 
+
+This implementation uses [tool calling](https://inspect.ai-safety-institute.org.uk/tools.html) to enable the model to execute bash commands. The flag values has been replaced with placeholders `REPLACE_ME_WITH_RANDOM_FLAG_N` to prevent dataset contaimination. Before running evaluation, replace these placeholders by running: 
+```bash
```

Comment:
Could we do this in Python on-demand as part of task initialisation? (e.g. perhaps in the `get_task_sample()` function?)

Reply:
Should we also add a cleanup step that reverts the flags back to placeholders after the task is done?

Reply:
Yes, I think that would be good! The `Plan` class has a `cleanup` function where this can be done.

Reply:
I've realized that the flags are baked into the Docker images, and the compose.yaml file pulls these images from Docker hub. If we were to generate random strings on the fly, that means the user would need to build the Docker images locally after the string gets generated, and I don't think this even can be done if the string generation was part of the task setup process. 

Alternatively, I think I can create a hash key and generate the hashed strings as flags in `read_dataset`. This way we don't even need to have a clean up step because the files were never modified, but we'll need to store the hash key in the code (or task.json). Would that be okay? Curious to see if you have any better suggestions, thanks!

Reply:
After trying a few different approaches, I'm thinking of getting the flag on the fly using a solver, but I'm stuck on changing the target, is there any way to access the `Target` in the `plan`? Thanks!

Reply:
What you could do here is read the target into the "store" (https://inspect.ai-safety-institute.org.uk/agents-api.html#sharing-state) in a solver, and then access that store value from within your scorer.

Reply:
i.e. you might just need a custom scorer for this.

Reply:
I can definitely make it work with a custom scorer, however in the logs the `Target` would still be the flag placeholder because I'm reading in the actual flag value in a solver by checking the flag file in the task environment container, would that be okay? I didn't find a way to modify the `Target` after it has been set in `read_dataset`.

Reply:
In this case there would be no target at all (so the logs would show no target) as this is synthesized at runtime.